src/backend/distributed/operations/repair_shards.c

```diff
@@ -1184,7 +1220,28 @@ CopyShardCommandList(ShardInterval *shardInterval, const char *sourceNodeName,
 
 	copyShardToNodeCommandsList = list_concat(copyShardToNodeCommandsList,
 											  tableRecreationCommandList);
+	return copyShardToNodeCommandsList;
+}
 
+
+/*
+ * CopyShardContentsCommandList generates command list to copy the data of the
+ * given shard placement from the source node to the target node. This copying
+ * requires a precreated table for the shard on the target node to have been
+ * created already (using RecreateShardDDLCommandList). Caller could optionally
+ * skip copying the data by the flag includeDataCopy.
+ *
+ * NOTE: The returned list also includes the post-load table creation commands.
+ */
+static List *
+CopyShardContentsCommandList(ShardInterval *shardInterval, const char *sourceNodeName,
+							 int32 sourceNodePort, bool includeDataCopy)
```

Comment:
I find the `includeDataCopy` argument a bit funky at the moment. If smells like a overload of the function. Already from before this refactor, but now even more.

It feels like there are 3 phases
 - create table
 - copy data
 - create indices

From the steps above step 2 is optional based on the `includeDataCopy` argument.

The above abstraction fails due to passing the boolean through to this function.

Instead I feel like we should make it more clear at a higher layer that these 3 phases are separate, and interpret the bool there to not call into the `CopyShardContentsCommandList` and instead go to the GetPostLoadTableCreationCommands` function that is now called as a side effect of the `CopyShardContentsCommandList`.